minidump-processor/src/op_analysis.rs

```diff
@@ -330,46 +622,116 @@ mod amd64 {
+            decoded_instruction: Instruction,
         ) -> Result<(), OpAnalysisError> {
-            let mut push_indirect_access = |address| {
-                accesses.push(MemoryAccess {
+            let mut push_implicit_access = |address, access_type| {
+                accesses.push(MemoryAddressInfo {
                     address,
                     is_likely_null_pointer_dereference: address == 0,
                     is_likely_guard_page: false,
-                    size: Some(1),
+                    size: Some(1), // TODO: correct size?
+                    access_type,
                 });
             };
 
+            let Some(opcode) = CommonOpcode::from_amd64_opcode(decoded_instruction.opcode()) else {
+                return Ok(());
+            };
+            match opcode {
+                CommonOpcode::CALL | CommonOpcode::PUSH => {
+                    if let Ok(rsp) = self.context.get_regspec(RegSpec::rsp()) {
+                        push_implicit_access(rsp, Some(OperandAccessType::Write));
+                    }
+                }
+                CommonOpcode::IRET
+                | CommonOpcode::IRETD
+                | CommonOpcode::IRETQ
+                | CommonOpcode::POP
+                | CommonOpcode::RETF
+                | CommonOpcode::RETURN => {
+                    if let Ok(rsp) = self.context.get_regspec(RegSpec::rsp()) {
+                        push_implicit_access(rsp, Some(OperandAccessType::Read));
+                    }
+                }
+                CommonOpcode::ADD
+                | CommonOpcode::CMP
+                | CommonOpcode::DEC
+                | CommonOpcode::INC
+                | CommonOpcode::JMP
+                | CommonOpcode::JMPF
+                | CommonOpcode::MOV
+                | CommonOpcode::MOVAPS
+                | CommonOpcode::MOVUPS
+                | CommonOpcode::SUB => (),
+            }
+            Ok(())
+        }
+    }
+
+    impl<'a> GetInstructionPointerUpdate<'a> {
+        pub fn new(
+            context: &'a MinidumpContext,
+            memory_list: Option<&'a minidump::UnifiedMemoryList<'a>>,
+            stack_memory: Option<minidump::UnifiedMemory<'a, 'a>>,
+        ) -> Self {
+            GetInstructionPointerUpdate {
+                context,
+                memory_list,
+                stack_memory,
+            }
+        }
+
+        /// Determine the update to instruction pointer implied by the given instruction, context
+        /// and memory
+        pub fn get_rip_update_from_instruction(
+            self,
+            decoded_instruction: Instruction,
+        ) -> Result<Option<MemoryAddressInfo>, OpAnalysisError> {
+            let rip_update = |address| {
+                Some(MemoryAddressInfo {
+                    address,
+                    is_likely_null_pointer_dereference: address == 0,
+                    is_likely_guard_page: false,
+                    size: Some(1),
```

Comment:
See -- The fact that there is nothing reasonable for you to put in these two fields is part of the reason why I think a split between `MemoryAddressInfo` and `MemoryAccess` is needed.